c10/core/SafePyObject.h

```diff
@@ -55,6 +55,22 @@ struct C10_API SafePyObject {
   c10::impl::PyInterpreter* pyinterpreter_;
 };
 
+// A newtype wrapper around SafePyObject for type safety when a python object
+// represents a specific type. Note that `T` is only used as a tag and isn't
+// actually used for any true purpose.
+template <typename T>
+struct C10_API SafePyObjectT : private SafePyObject {
+  SafePyObjectT(PyObject* data, c10::impl::PyInterpreter* pyinterpreter)
+      : SafePyObject(data, pyinterpreter) {}
+  SafePyObjectT(SafePyObjectT&& other) noexcept : SafePyObject(other) {}
```

Comment:
Curious, can't you just inherit them via `using SafePyObject::SafePyObject(SafePyObject&&)`?

Reply:
I wanted to make sure that you couldn't accidentally turn an untyped SafePyObject into a typed SafePyObjectT.